AzureMonitorAgent/agent.py

```diff
@@ -39,6 +39,7 @@
 import hashlib
 import fileinput
 import contextlib
+import distro
```

Comment:
Sorry for the drive-by comments @marcurdy  - I don't work on this project but I would like to get this issue resolved so my team can use Ubuntu 24.04 hosts on Azure without issue!

distro is a [third-party package](https://pypi.org/project/distro/), and from what I can see in the way that this AzureMonitorAgent project is [set up](https://github.com/Azure/azure-linux-extensions/tree/master/AzureMonitorAgent), the project has no requirements.txt / setup.py or any other mechanism to declare third-party dependencies. So I think this is intended to be deployed as just a single script, and adding a third-party dep will not work?

Reply:
on my ubuntu 24.04, distro comes ootb even in minimized installations. IMHO dropping it would be wrong, but it would be prudent to put it behind a try clause.